crypto/property/property_query.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal/propertyerr.h"
+#include "internal/property.h"
+#include "property_local.h"
+
+const OSSL_PROPERTY_DEFINITION *
+ossl_property_find_property(const OSSL_PROPERTY_LIST *list,
+                            OSSL_LIB_CTX *libctx, const char *name)
+{
+    OSSL_PROPERTY_IDX name_idx;
+    int i;
+
+    if (list == NULL || name == NULL
+        || (name_idx = ossl_property_name(libctx, name, 0)) == 0)
+        return NULL;
+
+    for (i = 0; i < list->n; i++)
```

Comment:
Property lists are _sorted_, a _binary search_ would be more appropriate here.

Reply:
For something that's not likely to be more than [wet finger in the air] 5-ish properties, that seems a bit overkill.

Reply:
This will come back and hurt at some point :)

Reply:
I see other functions that would benefit from the same thinking, for example `ossl_property_is_enabled()`...  so I take it this is for another PR :wink:

Reply:
That one was @slontis's doing...

Yeah, another PR is okay.

Reply:
I've also fixed it :)

#15614